src/common/sharedlibrary/CMakeLists.txt

```diff
@@ -14,6 +14,24 @@
 #
 # Authored by: Alexandros Frantzis <alexandros.frantzis@canonical.com>
 
+include(CheckFunctionExists)
+
+list(APPEND CMAKE_REQUIRED_LIBRARIES dl)
+list(APPEND CMAKE_REQUIRED_DEFINITIONS -D_GNU_SOURCE)
+
+# CMake docs suggests that we should, instead, be using check_symbol_exists.
+# The only problem there is that check_symbol_exists("dlvsym", "dlfcn.h", HAS_DLVSYM) never finds
+# the dlvsym symbol ☹.
+check_function_exists("dlvsym" HAS_DLVSYM)
```

Comment:
This doesn't seem to be working.

On my 18.04 system (with `dlvsym()`) I see the following cmake output:
```plain
CMake Warning at src/common/sharedlibrary/CMakeLists.txt:29 (message):
  dlvsym() not supported by libc.  Mir may attempt to load ABI-incompatible
  platform modules
```

Reply:
The cmake parts in my PR (#683) is tested to detect dlvsym on Arch and not detect it on Alpine